pkg/process/checks/config.go

```diff
@@ -0,0 +1,58 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package checks
+
+import (
+	"sync"
+
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	maxBatchSizeLogOnce         sync.Once
+	maxCtrProcsBatchSizeLogOnce sync.Once
+)
+
+// getMaxBatchSize returns the maximum number of items (processes, containers, process_discoveries) in a check payload
+func getMaxBatchSize() int {
+	batchSize := ddconfig.Datadog.GetInt("process_config.max_per_message")
+
+	if batchSize <= 0 {
+		batchSize = ddconfig.DefaultProcessMaxPerMessage
+		maxBatchSizeLogOnce.Do(func() {
+			log.Warnf("Invalid item count per message (<= 0), using default value of %d", ddconfig.DefaultProcessMaxPerMessage)
+		})
+
+	} else if batchSize > ddconfig.DefaultProcessMaxPerMessage {
+		batchSize = ddconfig.DefaultProcessMaxPerMessage
+		maxBatchSizeLogOnce.Do(func() {
+			log.Warnf("Overriding the configured max of item count per message because it exceeds maximum limit of %d", ddconfig.DefaultProcessMaxPerMessage)
+		})
+	}
+
+	return batchSize
+}
```

Comment:
This `sync.Once` is great! But how about using it to just set the value we want once? 

I'm thinking something like this:
```
var(
	maxBatchSizeOnce sync.Once
	maxBatchSize int
)

func getMaxBatchSize() int {
	maxBatchSizeOnce.Do(func() {
		batchSize := ddconfig.Datadog.GetInt("process_config.max_per_message")
		if batchSize <= 0 {
			batchSize = ddconfig.DefaultProcessMaxPerMessage
			log.Warnf("Invalid item count per message (<= 0), using default value of %d", ddconfig.DefaultProcessMaxPerMessage)

		} else if batchSize > ddconfig.DefaultProcessMaxPerMessage {
			batchSize = ddconfig.DefaultProcessMaxPerMessage
			log.Warnf("Overriding the configured max of item count per message because it exceeds maximum limit of %d", ddconfig.DefaultProcessMaxPerMessage)
		}
		maxBatchSize = batchSize
	})
	return maxBatchSize
}
```

Reply:
This seems a great idea! I didn't want to do it at first because this would block us from updating the setting during runtime. But this is not possible right now so I think that it's ok to implement it as you're suggesting and update it if we need to.

Reply:
wdyt?
https://github.com/DataDog/datadog-agent/pull/10698/commits/4da3f17de7ead5a83180da9b529ed3934f7e3808